src/python/pants/option/parser.py

```diff
@@ -424,15 +411,6 @@ def error(
                 error(OptionNameDoubleDash, arg_name=arg)
 
         # Validate kwargs.
-        deprecated_conditional(
-            lambda: "implicit_value" in kwargs,
-            removal_version="2.21.0a0",
```

Comment:
Just noting that I could imagine we do one two more 2.21.0.devX releases before a0 (hopefully not, though!), so this `removal_version` might end up being a slight fib. Fine by me, though, given we theorise it's unlikely people are using this.